x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -568,6 +585,85 @@ func readYaml(f string, cfg *setupConfig) error {
 	return c.Unpack(cfg)
 }
 
+func logToStderr(cfg *configuration.Configuration) {
+	// container forces logging to stderr
+	cfg.Settings.LoggingConfig.ToStderr = true
+	cfg.Settings.LoggingConfig.ToFiles = false
+}
+
+func handlePaths(pathCfg agentPathsConfig) error {
+	if pathCfg.State == "" {
+		return errors.New("STATE_PATH cannot be set to an empty string")
+	}
+	// ensure that the directory and sub-directory data exists
+	if err := os.MkdirAll(filepath.Join(pathCfg.State, "data"), 0755); err != nil {
+		return fmt.Errorf("preparing STATE_PATH(%s) failed: %s", pathCfg.State, err)
+	}
+	// ensure that the elastic-agent.yml exists in the state directory
+	if _, err := os.Stat(filepath.Join(pathCfg.State, paths.DefaultConfigName)); os.IsNotExist(err) {
+		if err := copyFile(filepath.Join(pathCfg.State, paths.DefaultConfigName), paths.ConfigFile(), 0); err != nil {
+			return err
+		}
+	}
+	// sync the downloads to the data directory
+	if err := syncDownloads(pathCfg.State); err != nil {
+		return fmt.Errorf("syncing download directory to STATE_PATH(%s) failed: %s", pathCfg.State, err)
+	}
+	paths.SetTop(filepath.Join(pathCfg.State, "data"))
+	paths.SetConfig(pathCfg.State)
```

Comment:
Can you make this configurable independently of the `pathCfg.State`?

Reply:
We could? What is the use case? Don't like to add an ENV unless it is useful.

Reply:
We have a usecase with an existing directory structure:
```
/app
/app/config
/app/logs
/app/data
```
When setting `STATE_PATH=/app` the agent expects the configuration at `/app/`.